pkg/networkplugin-syncer/handlers/ovn/handler.go

```diff
@@ -0,0 +1,112 @@
+package ovn
+
+import (
+	"errors"
+
+	goovn "github.com/ebay/go-ovn"
+	clientset "k8s.io/client-go/kubernetes"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/networkplugin-syncer/handlers/ovn/nbctl"
+)
+
+type SyncHandler struct {
+	event.HandlerBase
+	k8sClientset     clientset.Interface
+	nbctl            *nbctl.NbCtl
+	nbdb             goovn.Client
+	sbdb             goovn.Client
+	localEndpoint    *submV1.Endpoint
+	remoteEndpoints  map[string]*submV1.Endpoint
+	lastOvnGwChassis string
+}
+
+func (ovn *SyncHandler) GetName() string {
+	return "ovn-sync-handler"
+}
+
+func (ovn *SyncHandler) GetNetworkPlugin() string {
+	return "OVNKubernetes"
+}
+
+func NewSyncHandler(k8sClientset clientset.Interface) *SyncHandler {
+	return &SyncHandler{
+		remoteEndpoints: make(map[string]*submV1.Endpoint),
+		k8sClientset:    k8sClientset,
+	}
+}
+
+func (ovn *SyncHandler) Init() error {
+	if err := ovn.initClients(); err != nil {
+		return err
+	}
+
+	if err := ovn.ensureSubmarinerInfra(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ovn *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	ovn.localEndpoint = endpoint
+
+	return ovn.updateGatewayNode()
+}
+
+func (ovn *SyncHandler) LocalEndpointUpdated(endpoint *submV1.Endpoint) error {
+	ovn.localEndpoint = endpoint
+
+	return ovn.updateGatewayNode()
+}
+
+func (ovn *SyncHandler) LocalEndpointRemoved(endpoint *submV1.Endpoint) error {
+	if ovn.localEndpoint.Name == endpoint.Name {
+		ovn.localEndpoint = nil
+	}
+
+	return ovn.updateGatewayNode()
```

Comment:
Good one, I will avoid calling it then as ... it will make no difference. Without endpoint there's nothing special/or that we could do about the packets.